azerothcore/azerothcore-wotlk
# fix(CORE/Shutdown): Added checks to see if MapUpdater threads are joineable

## Changes Proposed:
-  Added a check to not join threads that are in invalid state/already finished
-  Only deactivate the MapUpdater on destructor only if it is active
- Moved a error log to coredebug as mentioned in the PR it is not providing any necessary information

## Issues Addressed:

Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7319

## Tests Performed:

- Compiled without problems
- Started & closed the server
- Had server open with character inside for +10mins


## How to Test the Changes:

1. Compile world and auth server
2. Start auth and world server
3. Press Control+C in world and auth server and notice no Segmentation fault is issued


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
